### ISSUE TITLE
D.C. (or D.S.) al Fine marking right before a section break causes the pause between sections to happen at the wrong time

### ISSUE THREAD

**Issue by BoBogus** [musescore/MuseScore#24832]
### Issue type

General playback bug

### Description with steps to reproduce

1. Create a new score with any instrumentation.
2. Add a section break at any given measure.  Make sure it is set to pause for some duration between sections.
3. On the same measure as the section break, add a D.S. al Coda or D.C. al Coda.
4. If using D.S. al Coda, add a segno in an earlier measure.

Expected behavior: On reaching the D.S. or D.C. marking, it should immediately jump to the capo or segno and begin playing back from there.  Then, on reaching the section break, it should pause as selected before starting the next section.

Actual behavior: On reaching the D.S. or D.C. marking, the cursor jumps to the appropriate place, but then unexpectedly pauses as though it's about to start the next section.  After the pause, it plays from the capo or segno as originally intended.  Finally, when it reaches the section break, it immediately begins playing the next section without a pause.

### Supporting files, videos and screenshots

[Broken Section Pauses.zip](https://github.com/user-attachments/files/17072156/Broken.Section.Pauses.zip)

### What is the latest version of MuseScore Studio where this issue is present?

OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.2-242570931, revision: github-musescore-musescore-3130f97

### Regression

I was unable to check

### Operating system

Windows 10 Pro, 64-bit

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by Jojo-Schmitz** [comment]
Duplicate of #14262 , I believe

**Comment by BoBogus** [comment]
I guess this is an additional case, since that appears to mimic what I'm seeing.  I also see that it's a long-standing issue.

Something that has occurred to me is that the section break isn't relevant until after the repeat or jump has occurred.  Out of curiosity, does MuseScore handle repeats and section breaks in any particular order on measures where both are present?

**Comment by jeetee** [comment]
The data structure that unrolls the playback order processes repeats and jumps before section breaks. However the part of code that turns that information into playback instructions hasn't been updated to process that subset of the information yet..

**Comment by BoBogus** [comment]
Interesting.  Are there any plans at this time to update it?

**Comment by Jojo-Schmitz** [comment]
Up to @jeetee, the master of repeats 😉

**Comment by jeetee** [comment]
> Up to [@jeetee](https://github.com/jeetee), the master of repeats 😉

Not quite; I'm not familiar with the playbackevents generating side of the code at all. But I can happily assist anyone that can work on this by walking them through which properties to consider when walking the repeatlistsegments.
